astropy/stats/bayesian_blocks.py

```diff
@@ -289,15 +289,15 @@ def compute_ncp_prior(self, N):
         If ``ncp_prior`` is not explicitly defined, compute it from ``gamma``
         or ``p0``.
         """
-        if self.ncp_prior is not None:
-            return self.ncp_prior
-        elif self.gamma is not None:
+
+        if self.gamma is not None:
             return -np.log(self.gamma)
         elif self.p0 is not None:
             return self.p0_prior(N)
         else:
-            raise ValueError("``ncp_prior`` is not defined, and cannot compute "
-                             "it as neither ``gamma`` nor ``p0`` is defined.")
+            raise ValueError("``ncp_prior`` is not defined, and cannot "
```

Comment:
now that you're removed setting to the defined value, could you also update the error message here, as well as update the short docstring above?

Reply:
Thanks for catching that.  I fixed it.